src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/ExplicitInitializationCheck.java

```diff
@@ -83,19 +95,33 @@ public void visitToken(DetailAST ast) {
                 && exprStart.getType() == TokenTypes.LITERAL_NULL) {
                 log(ident, MSG_KEY, ident.getText(), "null");
             }
+            if (!onlyObjectReferences) {
+                validateNonObjects(ast);
+            }
```

Comment:
General validation wouldn't work would it? Because it only will go into the condition if it is a `null` assignment. It will never check for other explicit initializations.